lxd/seccomp/seccomp.go

```diff
@@ -1730,15 +1730,27 @@ func (s *Server) HandleSysinfoSyscall(c Instance, siov *Iovec) int {
 	}
 
 	// Get instance uptime.
-	f, err := os.Stat(fmt.Sprintf("/proc/%d", siov.msg.init_pid))
+	pidStat, err := ioutil.ReadFile(fmt.Sprintf("/proc/%d/stat", siov.msg.init_pid))
 	if err != nil {
 		l.Warn("Failed getting init process info", logger.Ctx{"err": err, "pid": siov.msg.init_pid})
 		C.seccomp_notify_update_response(siov.resp, 0, C.uint32_t(seccompUserNotifFlagContinue))
 
 		return 0
 	}
 
-	instMetrics.Uptime = int64(time.Now().Sub(f.ModTime()).Seconds())
+	fields := strings.Fields(string(pidStat))
+	tickValue, err := strconv.ParseInt(fields[21], 10, 64)
+	if err != nil {
+		l.Warn("Failed parsing init process info", logger.Ctx{"err": err, "pid": siov.msg.init_pid})
+		C.seccomp_notify_update_response(siov.resp, 0, C.uint32_t(seccompUserNotifFlagContinue))
+
+		return 0
+	}
+
+	age := float64(tickValue / 100)
```

Comment:
Is that `100` value the same as `getconf CLK_TCK`? If yes, I don't know how risky it is to hardcode it? I check on amd64 and arm64 Ubuntu machine and both give `100` but that's a small sample.

Reply:
Yeah, it's the same. I looked at a few Go packages to get that value and they effectively all hardcode it to 100 as fetching the correct value requires an expensive cgo call.

It's fine as there are only two architectures that apparently have it set to something other than 100 and neither of those architectures support Go (IIRC it was ia64 and hppa but I may misremember).